### PR TITLE
Improved LoRA finetuning

### LINKED ISSUE #77: Question: about left padding
if the model is padded to the left and its a casual language model does that mean that padding tokens will receive attention from the rest of the sequence?

Should there me a mask to prevent tokens to attend to padding?

### PR DESCRIPTION
**Pull request by awaelchli**
Small fixes carried over from finetune_adapter.py
- right padding
- remove redundant gradient clipping
- avoid tensor cloning warning from torch

Fixes #77

### CODE REVIEW

**Review by lantiga — APPROVED** [review]
Looks good!

**Review by carmocca — APPROVED** [review]
